proxmoxtf/resource/container.go

```diff
@@ -2363,6 +2385,48 @@ func containerUpdate(ctx context.Context, d *schema.ResourceData, m interface{})
 		rebootRequired = true
 	}
 
+	if d.HasChange(mkResourceVirtualEnvironmentContainerFeatures) {
+		featuresBlock, err := structure.GetSchemaBlock(
+			resource,
+			d,
+			[]string{mkResourceVirtualEnvironmentContainerFeatures},
+			0,
+			true,
+		)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		nesting := types.CustomBool(featuresBlock[mkResourceVirtualEnvironmentContainerFeaturesNesting].(bool))
+		keyctl := types.CustomBool(featuresBlock[mkResourceVirtualEnvironmentContainerFeaturesKeyControl].(bool))
+		fuse := types.CustomBool(featuresBlock[mkResourceVirtualEnvironmentContainerFeaturesFUSE].(bool))
+		mountTypes := featuresBlock[mkResourceVirtualEnvironmentContainerFeaturesMountTypes].([]interface{})
+
+		features := containers.CustomFeatures{}
+
+		if nesting {
+			features.Nesting = &nesting
+		}
+
+		if keyctl {
+			features.KeyControl = &keyctl
+		}
+
+		if fuse {
+			features.FUSE = &fuse
+		}
+
+		if mountTypes != nil {
```

Comment:
I think we need an extra case for `mountType == nil`, when removing all mount types from the `features`

Reply:
Actually it's better to explicitly pass empty features list to clear the list. So, not specifying (`nil`) means that terraform doesn't touch `features` field and user can change it from UI.

Reply:
That would be against Terraform best practices. If a resource is added to a TF state, then it should only be managed by TF.

That also makes attribute "asymmetrical"; to add a flag we're adding the attribute and a list item in it. To revert this change we're removing only the list item, but not the attribute.

But I see your point. There is a potential issue with rolling out this feature -- it will overwrite mount types that people may have configured in their containers manually. So, updating these container will also require a manual step of updating their templates. I'll add a note to Release Notes for the next version to clarify that.

What do you think?

Reply:
Done!